graphql_compiler/compiler/ir_lowering_common.py

```diff
@@ -53,30 +53,34 @@ def to_match(self):
         return mark_name
 
 
-def lower_context_field_existence(ir_blocks):
+def lower_context_field_existence(ir_blocks, query_metadata_table):
     """Lower ContextFieldExistence expressions into lower-level expressions."""
     def regular_visitor_fn(expression):
         """Expression visitor function that rewrites ContextFieldExistence expressions."""
         if not isinstance(expression, ContextFieldExistence):
             return expression
 
+        location_type = query_metadata_table.get_location_info(expression.location).type
+
         # Since this function is only used in blocks that aren't ConstructResult,
         # the location check is performed using a regular ContextField expression.
         return BinaryComposition(
             u'!=',
-            ContextField(expression.location),
+            ContextField(expression.location, location_type),
             NullLiteral)
 
     def construct_result_visitor_fn(expression):
         """Expression visitor function that rewrites ContextFieldExistence expressions."""
         if not isinstance(expression, ContextFieldExistence):
             return expression
 
+        location_type = query_metadata_table.get_location_info(expression.location).type
```

Comment:
Would it be worth making this type grab a method proper on the `query_metadata_table`? It's used at least 3 times that I see.

Reply:
After thinking about it for a bit, I think saving a `.type` at the end of the existing function call is not worth defining a new function call on `query_metadata_table` -- it would expand the contract of the metadata table and mean we have to test a larger surface area, for comparatively little benefit. Let me know if you think I'm missing something or if you feel strongly about it.